github.cake

```diff
@@ -0,0 +1,78 @@
+#addin "nuget:https://api.nuget.org/v3/index.json?package=Octokit&version=0.48.0"
+#addin "nuget:https://api.nuget.org/v3/index.json?package=NuGet.Versioning&version=5.7.0"
+#addin "nuget:https://api.nuget.org/v3/index.json?package=Cake.Yaml&version=3.1.1"
```

Comment:
Cake.Yaml addin has already been added in the  `build.cake` file. Addins should only be added once, otherwise it could lead to a lot of confusion if different versions are used. I would add them all to the `build.cake` file to have it in a single place (or in a separate `addins.cake` file).

Reply:
Interesting. I just followed the prior art of [nuget.cake](https://github.com/cake-build/website/blob/79122cbfde83fa81334a7e3576c0acb435137cec/nuget.cake#L1-L3). Should we do the same with that one, then?

One nice thing about having the addins declared in the separate Cake file is the ability to run it directly (via dotnet cake) without having to run any task in build.cake, which is useful during development/testing.

If we had an `addins.cake` as you suggest, that is included by all the other files i.e. `build.cake`, `nuget.cake`, and `github.cake`, would that be a better approach? I'm assuming Cake would be smart enough to include the file only once, but haven't tested it does.

Reply:
`nuget.cake` only added addins, there are no duplications between `nuget.cake` and `build.cake`. The main issue is that addin is now added twice. The different `*.cake` files are basically just concatenated to a single file before they are processed by Roslyn. It's not that they have a separate scope like e.g. classes. If an addin is already loaded in `build.cake` it is also available in every other `.cake` file. The addins can be added to different files, but should not be added twice. At a certain complexity I personally find it easier to maintain if they are all in a single place (either the main cake file or a separate file). If we add an `addin.cake` it just needs to be loaded once in `build.cake`.

Reply:
> If we add an `addin.cake` it just needs to be loaded once in build.cake

I understand that this is all that is needed for the build process to work due to the concatenation of files, etc. but if one opens `github.cake` on VS Code, etc. would they get proper intellisense without declaring the addins that this file relies on? (I would think not, which is why I suggested a shared `addins.cake` would be added to all files instead of only `build.cake`)

I did a [quick test](https://github.com/augustoproiete-sandbox/test-multiple-cake-includes) and Cake is smart enough to #load a file only once, so that seems viable (?)

Reply:
> but if one opens github.cake on VS Code, etc. would they get proper intellisense without declaring the addins that this file relies on?

Yes, this should work following [Enabling IntelliSense](https://cakebuild.net/docs/integrations/editors/vscode/intellisense#enabling-intellisense), especially that the folder needs to be opened in VSCode and not a single file. In this case all Cake files will be concatenated and passed to OmniSharp as one big string. There are also lot of other reasons why IntelliSense for a single file won't work (e.g. global variables).